<launch>
  <arg name="filter_config" default="$(find brt)/config/rbc_particle_filter_tracker.yaml"/>
  <arg name="camera_config" default="$(find brt)/config/camera.yaml"/>
  <arg name="robot_config"  default="$(find brt)/config/robot_config.yaml"/>
  <arg name="camera" default="XTION" />

  <param name="use_sim_time" type="bool" value="true" />
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find arm_robot_model)/models/DarpaRobot.urdf.xacro'" />

  <!-- Pose Estimation Node-->
  <node name="rbc_particle_filter_robot_tracker" type="rbc_particle_filter_robot_tracker" pkg="brt" output="screen" respawn="false" launch-prefix="">
    <param name="robot_description_package_path" type="string" value="$(find arm_robot_model)" />
    <param name="camera_frame"  type="string"  value="$(arg camera)" />

    <rosparam command="load" file="$(arg filter_config)" />
    <rosparam command="load" file="$(arg camera_config)" />
    <rosparam command="load" file="$(arg robot_config)" />
  </node>

  <!-- This frame has to be published as the camera info in the bag files are relative to XTION_RGB (which should be XTION_IR) -->
  <!-- If this frame is not published, then the CameraDisplay in Rviz won't work-->
  <node pkg="tf" type="static_transform_publisher" name="$(arg camera)_IR_base_link1" args="0 0 0 0 0 0  $(arg camera) /$(arg camera)_RGB 100" />

</launch>
